starter-code/src/StackDataStructure.js

```diff
@@ -1,2 +1,40 @@
 function StackDataStructure () {
+    this.stackControl = [];
+    this.MAX_SIZE = 1;
 }
+
+StackDataStructure.prototype.isEmpty = function(){
+  if(this.stackControl.length == 0){
+    return true;
+  }else{
+    return false;
+  }
+
+};
+
+StackDataStructure.prototype.canPush = function(){
+  if(this.stackControl.length < this.MAX_SIZE){
+    return true;
+  }
+  else{
+    return false;
+  }
+
+};
+
+StackDataStructure.prototype.push = function(x){
+
+  if(this.stackControl.length >= this.MAX_SIZE){
+    return "Stack Overflow";}
+  this.stackControl.push(x);
+  return this.stackControl;
+
+};
+
+StackDataStructure.prototype.pop = function(){
```

Comment:
Prueba con este código:

`StackDataStructure.prototype.pop = function (element) {
  if (this.isEmpty()) {
    return "Stack Underflow";
  }

  return this.stackControl.pop();
};`

Aprovecha las funciones que ya tienes creadas ;)